go-runtime/encoding/encoding.go

```diff
@@ -31,37 +38,34 @@ func encodeValue(v reflect.Value, w *bytes.Buffer) error {
 		w.WriteString("null")
 		return nil
 	}
+
 	t := v.Type()
-	switch {
-	case t.Kind() == reflect.Ptr && t.Elem().Implements(jsonMarshaler):
-		v = v.Elem()
-		fallthrough
 
-	case t.Implements(jsonMarshaler):
-		enc := v.Interface().(json.Marshaler) //nolint:forcetypeassert
-		data, err := enc.MarshalJSON()
+	// Special-cased types
+	switch {
+	case t == reflect.TypeFor[time.Time]():
```

Comment:
Couldn't we just put a check for `ftl.Option` in here and use `json.Marshal`?

Reply:
You wouldn't do the check in the same way, you'd still do a prefix check but reuse the case block + json.Marshal. But that was answered elsewhere I think, re. the introduction of a custom encoder with the type registry.